src/telemetry/telemetryService.ts

```diff
@@ -83,12 +97,22 @@ export class TelemetryService {
       flushInterval: 10000, // 10 seconds is the default libraries' value.
     });
 
-    const segmentProperties = this.getTelemetryUserIdentity();
-    this._segmentAnalytics.identify(segmentProperties);
-    log.info('Segment analytics activated', segmentProperties);
+    this.userIdentity = await this.getTelemetryUserIdentity();
+    this._segmentAnalytics.identify(this.userIdentity);
+    this.isBufferingEvents = false;
+    log.info('Segment analytics activated', this.userIdentity);
+
+    // Process buffered events
+    if (this.eventBuffer.length > 0) {
```

Comment:
nit: This conditional doesn't do anything, right?

(alternatively: with these cases I definitely have a mild preference for `while (event = this.eventBuffer.shift())`, I know it doesn't make a real difference right now, but it's basically the same amount of syntax but provides greater resilience around potential future changes where the buffer or the `isBufferingEvents` flag could be changed concurrently to the loop)